be/src/runtime/stream_load/stream_load_pipe.h

```diff
@@ -119,18 +122,23 @@ class StreamLoadPipe : public MessageBodySink {
         while (bytes_read < *data_size) {
             if (_read_buf == nullptr || !_read_buf->has_remaining()) {
                 std::unique_lock<std::mutex> l(_lock);
-                _get_cond.wait(l, [&]() { return _cancelled || _finished || !_buf_queue.empty(); });
+                _get_cond.wait_for(l, std::chrono::seconds(3),
+                                   [&]() { return _cancelled || _finished || !_buf_queue.empty(); });
 
                 // cancelled
                 if (_cancelled) {
                     return _err_st;
                 }
                 // finished
                 if (_buf_queue.empty()) {
-                    DCHECK(_finished);
-                    *data_size = bytes_read;
-                    *eof = (bytes_read == 0);
-                    return Status::OK();
+                    if (_finished) {
+                        *data_size = bytes_read;
+                        *eof = (bytes_read == 0);
+                        return Status::OK();
+                    } else {
+                        *data_size = bytes_read;
+                        return Status::TimedOut("stream load time out");
```

Comment:
I think we should implement a new function read function for pipeline engine, since the old engine still need block read.